pkg/git/git.go

```diff
@@ -328,6 +329,20 @@ func (g *Git) gitCmd(output io.Writer, args ...string) error {
 		defer c.Close()
 	}
 
+	if len(g.knownHosts) != 0 {
+		f, err := ioutil.TempFile("", "known_hosts")
+		if err != nil {
+			return err
+		}
+		defer os.RemoveAll(f.Name())
+		defer f.Close()
+		if _, err := f.Write(g.knownHosts); err != nil {
```

Comment:
Don't you need to close after to the write to ensure everything is flushed to disk?